WMF Framework/Event Platform Client/EPCNetworkManager.swift

```diff
@@ -0,0 +1,140 @@
+
+import Foundation
+
+class EPCNetworkManager: EPCNetworkManaging {
+    
+    private let storageManager: EPCStorageManaging
+    private let session: Session
+    private let operationQueue: OperationQueue
+    
+    init(storageManager: EPCStorageManaging, session: Session = Session.shared) {
+        self.storageManager = storageManager
+        self.session = session
+        
+        operationQueue = OperationQueue()
+        operationQueue.maxConcurrentOperationCount = 1
+    }
+    
+    func httpPost(url: URL, body: NSDictionary) {
+        storageManager.createAndSavePost(with: url, body: body)
+    }
+    
+    func httpDownload(url: URL, completion: @escaping (Data?) -> Void) {
+        httpDownload(url: url, attempt: 0, maxAttempts: 5, attemptDelay: 2, completion: completion)
```

Comment:
@bearloga This is arbitrary logic I chose on config download retry attempts. Not sure who needs to OK this, maybe @jtminor ? This tries for 5 times, with 2 seconds in between attempts. Then it fails and the library will persist in the buffer until the next launch.

Reply:
That works! 5 attempts with 2 seconds between attempts sounds reasonable and generous :) We can run it by in the next BUOD<>Apps sync, but I doubt anyone would suggest a different configuration.

@linehan @dbrant: If we're assuming that the Android client will need to have same/similar behavior, do these settings for re-trying stream configuration retrieval make sense for y'all?

Reply:
This seems fine to me. We might re-architect this a bit down the road. I see this logic as being delegated to the implementation because it's part of fetching the information required to initialize the library (i.e. it takes place before the library 'comes into existence' in an operational sense). How the implementation gets to that starting line may be arbitrary. But, it's probably not a bad idea for them to match.